contracts/application/management/UnversionedAppManager.sol

```diff
@@ -0,0 +1,25 @@
+pragma solidity ^0.4.21;
+
+import "./BaseAppManager.sol";
+import "../versioning/ContractProvider.sol";
+import "../../upgradeability/UpgradeabilityProxyFactory.sol";
+
+contract UnversionedAppManager is BaseAppManager {
+  ContractProvider internal provider;
+
+  function UnversionedAppManager(ContractProvider _provider, UpgradeabilityProxyFactory _factory)
+    BaseAppManager(_factory)
+    public
+  {
+    setProvider(_provider);
+  }
+
+  function getProvider() internal view returns (ContractProvider) {
+    return provider;
+  }
+
+  function setProvider(ContractProvider _provider) public onlyOwner {
+    require(_provider != address(0));
```

Comment:
Avoid different type comparisson by casting like this:

`require((address(_provider) != address(0))`

or

`require(_provider != ContractProvider(0))`